plutus-ledger-constraints/src/Ledger/Constraints/OffChain.hs

```diff
@@ -424,15 +425,16 @@ data SomeLookupsAndConstraints where
 -- | Given a list of 'SomeLookupsAndConstraints' describing the constraints
 --   for several scripts, build a single transaction that runs all the scripts.
 mkSomeTx
-    :: [SomeLookupsAndConstraints]
+    :: Params
+    -> [SomeLookupsAndConstraints]
     -> Either MkTxError UnbalancedTx
-mkSomeTx xs =
+mkSomeTx params xs =
```

Comment:
I don't see any reason to export it aside testability. But as it is exposed, depracation and a new `mkSomeTxWithParams` function?

Reply:
IMHO we're still too much a beta level library to be doing this deprecation dance. Even for `mkTx` I find it dubious since its type changes, so nothing would silently break and the fix is pretty clear, but it is probably so widely used that it's worth doing the dance.

Reply:
Good point. Since a number of people have complained about us doing too much breaking changes in the API, I though deprecating would be the way to go, even though we're still a beta level library.

I understand that we need a deprecation policy and a well defined public API.